x-pack/plugins/apm/public/components/app/TransactionDetails/Transaction/WaterfallContainer/index.js

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { get } from 'lodash';
+import Waterfall from './Waterfall';
+import { getServiceColors } from './getServiceColors';
+import ServiceLegends from './ServiceLegends';
+import { loadTrace } from '../../../../../services/rest/apm';
+import { TRACE_ID } from '../../../../../../common/constants';
+import { getWaterfall } from './Waterfall/waterfall_helpers/waterfall_helpers';
+
+export class WaterfallContainer extends PureComponent {
```

Comment:
`PureComponent` makes me nervous in general but do you think there's a good trade-off for using it here? (Nervous because I know there are some ways to slip up with them and I'm not always sure there's an advantage to using it.)

Reply:
`PureComponent` is just `Component` with `shouldComponentUpdate` checking for prop changes (shallow equality). afaik it only causes problems if data is mutated (which we avoid anyway). It does add an extra check, that could reduce perf.

In this case the waterfall was being rendered several times necessarily. But I can remove it now, and then we can do some perf analysis later if you think.